### PR TITLE
CLOSED 268

### PR DESCRIPTION
**Pull request by niqt**
Add search box and search function for contats

### CODE REVIEW

File: src/qml/ContactListPage.qml [hunk 1]
```diff
@@ -10,8 +10,30 @@ Page {
     PageHeader {
         id: header
         text: qsTr("Contacts")
+        MouseArea {
+                    anchors.fill: parent
```
Comment (rburchell) [inline src/qml/ContactListPage.qml:14]: this indentation looks wrong. have you used tabs here?

File: src/qml/SearchBox.qml [hunk 2]
```diff
@@ -0,0 +1,138 @@
+/**
```
Comment (rburchell) [inline src/qml/SearchBox.qml:1]: this licensing header is completely unacceptable. you don't include the license text it refers to.

File: src/qml/SearchBox.qml [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+ * the distribution.
+ */
+
+import QtQuick 1.1
+import com.nokia.meego 1.0
+
+// A Custom made search box element, as there is no SearchBox in MeeGo Qt Quick
+// Components Extras. Some implementation ripped from the QQC Symbian
+// SearchBox.
+Item {
```
Comment (rburchell) [inline src/qml/SearchBox.qml:15]: for performance reasons, try not to use Item unnecessarily. it usually means your object heirachy can be simplified. I don't have any instant suggestions to mind, ask if you can't figure something out.

File: src/qml/SearchBox.qml [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+            function __handleIconSource(iconId) {
+                var prefix = "icon-m-"
+                // check if id starts with prefix and use it as is
+                // otherwise append prefix and use the inverted version if required
+                if (iconId.indexOf(prefix) !== 0)
+                    iconId =  prefix.concat(iconId).concat(theme.inverted ? "-white" : "");
+                return "image://theme/" + iconId;
+            }
+
+            clip: true
```
Comment (rburchell) [inline src/qml/SearchBox.qml:68]: why is this needed?

File: src/qml/SearchBox.qml [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+                property string __searchIconId: "toolbar-search"
+
+                anchors {
+                    left: parent.left
+                    verticalCenter: parent.verticalCenter
+                    margins: 10 * 2
+                }
+
+                smooth: true
```
Comment (rburchell) [inline src/qml/SearchBox.qml:103]: do these really look terrible without smoothing?

File: src/qml/api/ContactListWidget.qml [hunk 6]
```diff
@@ -61,7 +63,8 @@ ListView {
         id: emptyListView
         visible: parent.count == 0
         anchors.centerIn: parent
-        text: qsTr("You haven't added any contacts yet.")
+        text: ((searching) ? qsTr("Contact not found.") : qsTr("You haven't added any contacts yet."));
+        //qsTr("You haven't added any contacts yet.")
```
Comment (rburchell) [inline src/qml/api/ContactListWidget.qml:67]: if it isn't used, remove it.

**Comment by rburchell** [comment]
I'm reverting this, primarily because of the licensing issue, but also because of the smaller issues.

In future, please allow me to review work you're submitting to something I maintain, or at least make an effort to get in contact with me before merging it, so that issues like this can be avoided. I've been out of contact on holiday for the past few days (hence no review), but I don't see any hilights or email about this other than the automated ones from Github.

**Comment by niqt** [comment]
sorry, I think there was a misunderstanding in communication (I thought I had explained in chat, however I was not clear enough).
